x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/pipeline_json_badges.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import { EuiBadgeGroup, EuiBadge, EuiToolTip } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { DEFAULT_PIPELINE_NAME } from '../../../../../../common/constants';
+
+import { isManagedPipeline } from '../../../../shared/pipelines/is_managed';
+
+import { IndexPipelinesConfigurationsLogic } from './pipelines_json_configurations_logic';
+
+const ManagedPipelineBadge = (
```

Comment:
I think we should change them into React.FC components. If I remember correctly we had some re-rendering issues because of this const pattern.
There was [a PR ](https://github.com/elastic/kibana/pull/138496)that is changing those, not exactly same but similar.

I will not block the PR but if you can check and update these, that would be great.